### ISSUE TITLE
MIN MAX: allow cell references or min/max values of column

### ISSUE THREAD

**Issue by novoid** [tbanel/orgtblasciiplot#1]
Hi!

The following table works fine:

```
| level | seconds |             |  links |              |
|-------+---------+-------------+--------+--------------|
|     2 |       7 | ▏           |  46861 | ▊            |
|     3 |      27 | ▌           | 116671 | █▉           |
|     4 |     136 | ██▊         | 292039 | ████▋        |
|     5 |     547 | ███████████ | 733399 | ███████████▊ |
#+TBLFM: $3='(orgtbl-uc-draw-cont $2 0 600 12)::$5='(orgtbl-uc-draw-cont $4 0 750000 12)
```

Awesome feature, I really do thank you for providing that.

What would be even more awesome would be something like this:

```
| level | seconds |        |  links |        |
|-------+---------+--------+--------+--------|
|     2 |       7 | #ERROR |  46861 | #ERROR |
|     3 |      27 | #ERROR | 116671 | #ERROR |
|     4 |     136 | #ERROR | 292039 | #ERROR |
|     5 |     547 | #ERROR | 733399 | #ERROR |
#+TBLFM: $3='(orgtbl-uc-draw-cont $2 0 @>$2 12)::$5='(orgtbl-uc-draw-cont $4 min($4) max($4) 12)
```

Please note the `@>$2` and the very naïve `min()` and `max()` expressions.

This would ease the manual effort to find the local maximum and/or minimum of a column in order to visualize their distribution with your cool tool.

However, `@>$2` currently ends up in an error. Maybe there is at least a trick to get those min/max expressions working.

What do you think?

**Comment by tbanel** [comment]
Very good idea!
Here is a patch to make `@>$2` work:

      (defun orgtbl-ascii-draw (value min max &optional width characters)
        "Draw an ascii bar in a table.
      VALUE is the value to plot, it determines the width of the bar to draw.
      MIN is the value that will be displayed as empty (zero width bar).
      MAX is the value that will draw a bar filling all the WIDTH.
      WIDTH is the span in characters from MIN to MAX.
      CHARACTERS is a string that will compose the bar, with shades of grey
      from pure white to pure black.  It defaults to a 10 characters string
      of regular ascii characters."
        (let* ((width      (ceiling (or width 12)))
               (characters (or characters " .:;c!lhVHW"))
               (len        (1- (length characters)))
               (value      (float (if (numberp value)
                                      value (string-to-number value))))
               (min        (float (if (numberp min)  ;; <------ patch
                                      min (string-to-number min))))
               (max        (float (if (numberp max)  ;; <------ patch
                                      max (string-to-number max))))
               (relative   (/ (- value min) (- max min)))
               (steps      (round (* relative width len))))
          (cond ((< steps             0) "too small")
                ((> steps (* width len)) "too large")
                (t (let* ((int-division (/ steps len))
                          (remainder    (- steps (* int-division len))))
                     (concat (make-string int-division (elt characters len))
                             (string (elt characters remainder))))))))

This does not work with `min($4)`, though.

Note also that now `orgtblasciiplot` comes out of the box: it made its way into Org Mode core some time ago. I will see if I can submit a patch.

Thanks for this suggestion.
Thierry

**Comment by novoid** [comment]
Hi Thierry,

Thanks for the quick response!

I stumbled over your precious code via https://www.reddit.com/r/orgmode/comments/7k7tfy/preventing_orgmode_pdf_tables_from_running_off/ and found tons of use-cases in my environment.

Your patch works.

I wrote this issue on Github because I thought that this is the quickest way to the original developer ;-)

Looking forward to see the patch reaching orgmode/maint which I am using.

Have a nice new year's evening and greetings from Graz/Austria!
